AnaProd/anaTupleProducer.py

```diff
@@ -42,30 +54,78 @@ def addAllVariables(dfw, syst_name, isData, trigger_class, mode, nLegs):
         dfw.colToSave.extend(hltBranches)
     dfw.Define(f"Tau_recoJetMatchIdx", f"FindMatching(Tau_p4, Jet_p4, 0.5)")
     dfw.Define(f"Muon_recoJetMatchIdx", f"FindMatching(Muon_p4, Jet_p4, 0.5)")
     dfw.Define( f"Electron_recoJetMatchIdx", f"FindMatching(Electron_p4, Jet_p4, 0.5)")
-    dfw.DefineAndAppend("channelId","static_cast<int>(httCand.channel())")
+    dfw.DefineAndAppend("channelId","static_cast<int>(HttCandidate.channel())")
     if mode == "HH":
-        channel_to_select = " || ".join(f"httCand.channel()==Channel::{ch}" for ch in config["GLOBAL"]["channelSelection"])
+        channel_to_select = " || ".join(f"HttCandidate.channel()==Channel::{ch}" for ch in config["GLOBAL"]["channelSelection"])
         dfw.Filter(channel_to_select, "select channels")
-    jet_obs = []
-    jet_obs.extend(JetObservables)
+
+    fatjet_obs = []
+    fatjet_obs.extend(FatJetObservables)
     if not isData:
-        dfw.Define(f"Jet_genJet_idx", f" FindMatching(Jet_p4,GenJet_p4,0.3)")
-        jet_obs.extend(JetObservablesMC)
-        if "LHE_HLT" in dfw.df.GetColumnNames():
-            dfw.colToSave.append("LHE_HT")
+        dfw.Define(f"FatJet_genJet_idx", f" FindMatching(FatJet_p4[FatJet_bbCand],GenJetAK8_p4,0.3)")
+        fatjet_obs.extend(JetObservablesMC)
+    dfw.DefineAndAppend(f"SelectedFatJet_pt", f"v_ops::pt(FatJet_p4[FatJet_bbCand])")
+    dfw.DefineAndAppend(f"SelectedFatJet_eta", f"v_ops::eta(FatJet_p4[FatJet_bbCand])")
+    dfw.DefineAndAppend(f"SelectedFatJet_phi", f"v_ops::phi(FatJet_p4[FatJet_bbCand])")
+    dfw.DefineAndAppend(f"SelectedFatJet_mass", f"v_ops::mass(FatJet_p4[FatJet_bbCand])")
+    for fatjetVar in fatjet_obs:
+        if(f"FatJet_{fatjetVar}" not in dfw.df.GetColumnNames()): continue
+        dfw.DefineAndAppend(f"SelectedFatJet_{fatjetVar}", f"FatJet_{fatjetVar}[FatJet_bbCand]")
+    subjet_obs = []
+    subjet_obs.extend(SubJetObservables)
+    if not isData:
+        dfw.Define(f"SubJet1_genJet_idx", f" FindMatching(SubJet_p4[FatJet_subJetIdx1],SubGenJetAK8_p4,0.3)")
+        dfw.Define(f"SubJet2_genJet_idx", f" FindMatching(SubJet_p4[FatJet_subJetIdx2],SubGenJetAK8_p4,0.3)")
+        fatjet_obs.extend(SubJetObservablesMC)
+    for subJetVar in subjet_obs:
+        for subJetIdx in [1,2]:
+            dfw.DefineAndAppend(f"SubJet{subJetIdx}_{subJetVar}", f"""
+                                RVecF subjet_var;
+                                for(size_t fj_idx; fj_idx<SelectedFatJet_pt.size(); fj_idx++){{
+                                    for (size_t sj_idx; sj_idx<SubJet_{subJetVar}.size(); sj_idx++){{
+                                            if(sj_idx == SelectedFatJet_subJetIdx{subJetIdx}.at(fj_idx)){{
+                                                subjet_var.push_back(SubJet_{subJetVar}.at(sj_idx));
+                                            }}
+                                        }}
+                                    }}
+                                return subjet_var;
```

Comment:
```suggestion
RVecF subjet_var(SelectedFatJet_pt.size(), 0.f);
for(size_t fj_idx = 0; fj_idx<SelectedFatJet_pt.size(); fj_idx++) {{
  auto sj_idx == SelectedFatJet_subJetIdx{subJetIdx}.at(fj_idx);
  if(sj_idx >= 0 && sj_idx < SubJet_{subJetVar}.size())
    subjet_var[fj_idx] = SubJet_{subJetVar}.at(sj_idx);
}}
return subjet_var;
```

also add `SubJet{subJetIdx}_isValid` branch